terraform/aws/modules/ec2/main.tf

```diff
@@ -0,0 +1,12 @@
+provider "aws" {
+    access_key = "${var.access_key}"
+    secret_key = "${var.secret_key}"
+    region = "ap-south-1"
+}
+
+resource "aws_instance" "ec2_instance" {
+    ami = "${var.ami_id}"
+    count = "${var.number_of_instances}"
+    instance_type = "${var.instance_type}"
+    key_name = "${var.ami_key_pair_name}"
```

Comment:
pls add more conditions.
1) If its a public instance, create and attach an EIP
2) Specify the subnet
3) specify additional sec groups if needed (and create a default one)
4) specify the root disk parameters
5) add options to attach a secondary disk if needed.
6) associate a specified IAM role if specified.